src/harness/reference_models/aggregate_interference/aggregate_interference.py

```diff
@@ -0,0 +1,362 @@
+
+  # Get interference caused by each grant
+  for grant in neighborhood_grants:
+    cbsd_interference = interf.computeInterferenceFssBlocking(grant, constraint,
+                          fss_info, grant.max_eirp)
+    cbsd_interference_list.append(cbsd_interference)
+
+  aggregate_interference[constraint] = convertAndSumInterference(cbsd_interference_list)
+  return aggregate_interference
+
+
+def calculateAggregateInterferenceForFss(fss_record, cbsd_list):
+  """Calculates aggregate interference for FSS.
+  
+  Calculates aggregate interference for each protection constraint in the FSS
+  based on the interferences caused by CBSD's grants in the neighborhood of
+  the FSS protection entity.
+
+  Args:
+    fss_record: FSS protection entity
+    cbsd_list: list of CBSD objects containing RegistrationRequests and GrantRequests
+  Returns:
+    Aggregate interference to FSS in the dictionary format,where the key is a tuple. 
+    {(latitude, longitude, low_frequency, high_frequency,entity_type):
+    aggregate_interference(dBm)}
+  """
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+  # Get FSS T&C Flag value
+  fss_weight1 = fss_record['deploymentParam'][0]['weight1']
+  fss_weight2 = fss_record['deploymentParam'][0]['weight2']
+
+  fss_ttc_flag = fss_record['deploymentParam'][0]['ttc']
+  # Get the protection point of the FSS
+  fss_point = fss_record['deploymentParam'][0]['installationParam']
+  protection_point = (fss_point['latitude'], fss_point['longitude'])
+
+  # Get the frequency range of the FSS
+  fss_freq_range = fss_record['deploymentParam'][0]['operationParam']['operationFrequencyRange']
+  fss_low_freq = fss_freq_range['lowFrequency']
+  fss_high_freq = fss_freq_range['highFrequency']
+
+  # Get FSS information
+  fss_info = interf.FssProtectionPoint(latitude=fss_point['latitude'], 
+               longitude=fss_point['longitude'],
+               height_agl=fss_point['height'],
+               max_gain_dbi=fss_point['antennaGain'],
+               pointing_azimuth=fss_point['antennaAzimuth'],
+               pointing_elevation=fss_point['antennaElevationAngle'],
+               weight_1=fss_weight1, weight_2=fss_weight2)
+
+  aggregate_interference = {}
+  # FSS Passband is between 3600 and 4200
+  if(fss_low_freq >= interf.FSS_LOW_FREQ_HZ and 
+         fss_low_freq < interf.CBRS_HIGH_FREQ_HZ):
+
+    # Get channels for co-channel CBSDs
+    protection_channels = interf.getProtectedChannels(fss_low_freq, interf.CBRS_HIGH_FREQ_HZ)
+    aggregate_interference = aggregateInterferenceForPoint(protection_point, 
+                               protection_channels, fss_low_freq, 
+                               interf.CBRS_HIGH_FREQ_HZ, grant_list, fss_info, 
+                               None, interf.ProtectedEntityType.FSS_CO_CHANNEL)
+
+    # 5MHz channelization is not required for Blocking protection
+    # FSS in-band blocking algorithm
+    aggregate_interference = aggrFSSBlockingConstraint(
```

Comment:
Wrong. You are loosing the result of previous calculation here since you reassign the aggregate_interference variable.
What you probably intent is:
```
aggregate_interference.update(aggregateInterferenceForPomt(...))
```

Reply:
Updated   logic call aggregate interference module separately for FSS Co-channel and   FSS Blocking protection types